packages/core/contracts/cross-chain-oracle/chain-adapters/Optimism_ParentMessenger.sol

```diff
@@ -103,4 +103,11 @@ contract Optimism_ParentMessenger is CrossDomainEnabled, ParentMessengerInterfac
         ParentMessengerConsumerInterface(oracleHub).processMessageFromChild(childChainId, data);
         emit MessageReceivedFromChild(data, childMessenger, oracleHub);
     }
+
+    /**
+     * @notice sendMessageToChild never sends any msg.value, so this should always return 0.
+     */
+    function getL1CallValue() external pure override returns (uint256) {
```

Comment:
can you add this function to the `ParentMessengerBase` such that it defaults to 0 in all cases except when overrides, such as in the `Arbitrum_ParentMessenger.sol`

Reply:
In fact, can we not move a bunch of the `setChild*` functions into the `Base`?

Reply:
We can definitely move `getL1CallValue` but the other ones we cannot because the internal `sendMessage` implementations are different